pkg/pdp/aggregator/fns/fns_test.go

```diff
@@ -0,0 +1,205 @@
+			pieceSizes:         []int64{64 * MB},
+			expectedBufferSize: 0,
+			expectedAggCount:   1,
+		},
+		{
+			name: "Single piece just under 128MB => remains in buffer, no aggregates",
+			// 63MB unpadded rounds up to 64MB padded, but it's still a single piece.
+			pieceSizes:         []int64{63 * MB},
+			expectedBufferSize: 64 * MB, // after rounding
+			expectedAggCount:   0,
+		},
+		{
+			name: "Single piece exactly 128MB => triggers immediate aggregate",
+			// A piece right at the threshold gets flushed immediately.
+			pieceSizes:         []int64{128 * MB},
+			expectedBufferSize: 0,
+			expectedAggCount:   1,
+		},
+		{
+			name: "Single piece >128MB but <256MB => triggers immediate aggregate",
+			// Because newPiece.PaddedSize() > 128MB => aggregator flushes right away.
+			pieceSizes:         []int64{200 * MB},
+			expectedBufferSize: 0,
+			expectedAggCount:   1,
+		},
+		{
+			name: "Single piece 192MB => triggers immediate aggregate",
+			// By definition, a piece this large is also flushed right away as it pads to 256MB.
+			pieceSizes:         []int64{192 * MB},
+			expectedBufferSize: 0,
+			expectedAggCount:   1,
+		},
+		{
+			name: "Two small pieces that sum exactly 128MB => one aggregate, empty buffer",
+			pieceSizes: []int64{
+				63 * MB, // each might become ~64MB padded
+				63 * MB, // combined crosses threshold
+			},
+			// The aggregator hits >=128MB on the second piece => flushes => buffer resets.
+			expectedBufferSize: 0,
+			expectedAggCount:   1,
+		},
+		{
+			name: "Two small pieces slightly over 128MB => flush once, leftover in new buffer",
+			pieceSizes: []int64{
+				70 * MB, // each might become ~128MB padded
+				70 * MB, // second piece triggers flush, but check leftover logic
+			},
+			// Because the first piece alone is >128MB padded, it flushes immediately,
+			// leaving a fresh buffer that gets the second piece => second piece also
+			// flushes. The net effect is typically two immediate aggregates if 70MB
+			// rounds to 128MB. If you want them combined, pick sizes that together
+			// cross 128 but not individually.
+			expectedBufferSize: 0,
+			expectedAggCount:   2,
+		},
+		{
+			name: "Multiple pieces cause multiple flushes",
+			pieceSizes: []int64{
+				32 * MB, // ~64MB padded
+				32 * MB, // crosses 128 => flush #1
+				40 * MB, // alone remains in buffer since <128MB padded
+				90 * MB, // once buffer + 90 crosses threshold => flush #2
+				70 * MB, // new buffer, triggers flush #3
+			},
+			expectedBufferSize: 0,
+			expectedAggCount:   3,
+		},
+		{
+			name: "Single piece >256MB (if code permits) => immediate flush or error",
+			// TODO(forrest): do we expect this to be an error case?
+			// The aggregator code comment suggests >256MB is out-of-scope, but not
+			// strictly enforced. Some implementations could treat this
+			// as an error or just flush.
+			pieceSizes:         []int64{300 * MB},
+			expectedBufferSize: 0,
+			expectedAggCount:   1,
```

Comment:
Unsure if this case is realistic, `AggregatePieces` states a pieces should never be larger than 256MB, but doesn't enforce the constraint.